docs/sources/developers/plugins/add-authentication-for-data-source-plugins.md

```diff
@@ -0,0 +1,159 @@
++++
+title = "Add authentication for data source plugins"
+type = "docs"
+aliases = ["/docs/grafana/latest/plugins/developing/auth-for-datasources/"]
++++
+
+# Add authentication for data source plugins
+
+This page explains how to authenticate against an third-party API from a data source plugin using the the Grafana authentication proxy.
+
+When a user saves a password, or any other secret data, as a data source option, Grafana encrypts the data and stores it in the Grafana database. Encrypted data source options can only be decrypted on the Grafana server. In other words, once a password has been saved, no sensitive data will be sent to the browser.
+
+Since the browser can't decrypt secrets like passwords or API keys, any external requests that need to be authenticated, need to go through the Grafana server. To make this process easier, Grafana offers a built-in proxy that intercepts outgoing requests, decrypts any secrets, and forwards the request to the original URL. Any data source plugin that needs to authenticate against an external service, or third-party API, needs to use the Grafana _authentication proxy_.
```

Comment:
```suggestion
Since the browser can't decrypt secrets like passwords or API keys, any external requests that need to be authenticated, need to go through the Grafana server. To make this process easier, Grafana offers a built-in proxy that intercepts outgoing requests, decrypts any secrets, and forwards the request to the original URL. Any data source plugin that needs to authenticate against an external service, or third-party API, needs to use the Grafana data source proxy.
```